File: src/main/java/tvmaze/TvMaze.java
package tvmaze;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;
import rx.Observable;
import tvmaze.pojo.TvMazeEpisode;
import tvmaze.pojo.TvMazeSearch;

public interface TvMaze {
	@GET("search/shows")
	Call<List<TvMazeSearch>> getShows(@Query("q") String query);

	@GET("shows/{id}/episodes")
	Call<List<TvMazeEpisode>> getEpisodes(@Path("id") int id);
}

File: src/main/kotlin/extensions/TvMaze.kt
package extensions

import tvmaze.TvMaze
import tvmaze.pojo.TvMazeEpisode
import tvmaze.pojo.TvMazeSearch
import java.io.IOException
import java.util.*

fun TvMaze.getTvShowFromName(tvShow: String): Optional<List<TvMazeSearch>> {
	try {
		val response = this.getShows(tvShow).execute()
		if (!response.isSuccessful) {
			println(response.errorBody())
		} else {
			return Optional.of(response.body())
		}
	} catch (e: IOException) {
		e.printStackTrace()
	}
	return Optional.empty()
}

fun TvMaze.getEpisodes(showID: Int): Optional<List<TvMazeEpisode>> {
	try {
		val response = this.getEpisodes(showID).execute()
		if (!response.isSuccessful) {
			println(response.errorBody())
		} else {
			return Optional.of(response.body())
		}
	} catch (e: IOException) {
		e.printStackTrace()
	}
	return Optional.empty()
}